src/analysis/schwartz_space.lean

```diff
@@ -392,6 +395,41 @@ instance : uniform_add_group 𝓢(E, F) :=
 instance : locally_convex_space ℝ 𝓢(E, F) :=
 seminorm_family.to_locally_convex_space (schwartz_with_seminorms ℝ E F)
 
+instance : topological_space.first_countable_topology (𝓢(E, F)) :=
+(schwartz_with_seminorms ℝ E F).first_countable
+
 end topology
 
+section distribution
+
+variables (𝕜 F)
+variables [is_R_or_C 𝕜] [normed_space 𝕜 F] [smul_comm_class ℝ 𝕜 F]
+
+/-- The delta distribution as a linear map. -/
+def delta_aux (x₀ : E) : 𝓢(E, F) →ₗ[𝕜] F :=
+{ to_fun := λ f, f x₀,
+  map_add' := λ f g, add_apply,
+  map_smul' := λ a f, smul_apply }
+
+lemma delta_aux_apply (x₀ : E) (f : 𝓢(E, F)) : delta_aux 𝕜 F x₀ f = f x₀ := rfl
+
+/-- The delta distribution -/
+def delta (x₀ : E) : 𝓢(E, F) →L[𝕜] F :=
+{ cont :=
+  begin
+    refine (delta_aux 𝕜 F x₀).continuous_of_locally_bounded (λ s hs, _),
+    rw bornology.is_vonN_bounded_iff_seminorm_bounded (schwartz_with_seminorms 𝕜 E F) at hs,
+    rcases hs (0,0) with ⟨r, hr, hs'⟩,
+    rw normed_space.image_is_vonN_bounded_iff,
+    use r,
+    intros f hf,
+    rw [delta_aux_apply, ← @norm_iterated_fderiv_zero ℝ],
+    exact (norm_iterated_fderiv_le_seminorm 𝕜 f 0 x₀).trans (hs' f hf).le,
+  end,
```

Comment:
```suggestion
  begin
    change continuous (delta_aux 𝕜 F x₀),
    refine seminorm.continuous_from_bounded (schwartz_with_seminorms 𝕜 E F)
      (norm_with_seminorms 𝕜 F) _ (λ i, ⟨{0}, 1, one_ne_zero, λ f, _⟩),
    rw [finset.sup_singleton, one_smul],
    exact norm_le_seminorm 𝕜 f x₀
  end,
```

with 
```lean
lemma norm_le_seminorm (f : 𝓢(E, F)) (x₀ : E):
  ∥f x₀∥ ≤ (schwartz_map.seminorm 𝕜 0 0) f :=
begin
  have := schwartz_map.le_seminorm 𝕜 0 0 f x₀,
  rwa [norm_iterated_fderiv_zero, pow_zero, one_mul] at this,
end
```
added earlier in the file (I'd say after `norm_pow_mul_le_seminorm`)

Reply:
I think the bounded continuous function idea is good, so I changed everything to that - the proof for the continuity is almost the same as for the delta function, which I find very neat (it is to be expected of course).